### PR TITLE
fix: Decode alias-typed HTTP params recursively based on their underlying type

### PR DESCRIPTION
**Pull request by bmoylan**
This change is [Reviewable](https://reviewable.io/reviews/palantir/conjure-go/258)

### CODE REVIEW

File: conjure/serverwriter.go [hunk 1]
```diff
@@ -354,21 +352,28 @@ func astForDecodeHTTPParamInternal(methodBody *jen.Group, argName string, argTyp
 				g.Id(outVarName).Op("=").Append(jen.Id(outVarName), jen.Id("convertedVal"))
 			})
 		}
+	case *types.Set:
```
Comment (bmoylan) [inline conjure/serverwriter.go:355]: same as types.List case

Comment (tabboud) [inline conjure/serverwriter.go:355]: Can you add a unit test for sets in `serverwriter_test.go` since it looks like we don't have any yet.

**Review by tabboud — COMMENTED** [review]
Change looks good, just 1 request for another test case

File: conjure/serverwriter.go (hunk 1 above)
Comment (bmoylan) [reply conjure/serverwriter.go:355]: Yup, done!

**Review by tabboud — APPROVED** [review]

**Pull request merged**